### PR TITLE
Refactor MembershipForm UI (#367)

### LINKED ISSUE #367: Refactor MembershipForm UI
Right now, our sign in form looks like this:
image

And makes anyone trying to use our website feel like:
image

### How we can do this
1. Start with a drawing of suggested changes (use google to search for various examples)
1. Clear out the content of the render method in the `MembershipForm` component and add your improved work
1. Connect the front end code with the functions needed to sign up a user.

The code can be found in [`src/Pages/MembershipApplication/MembershipForm.js`](https://github.com/SCE-Development/Core-v4/blob/dev/src/Pages/MembershipApplication/MembershipForm.js).

### PR DESCRIPTION
**Pull request by IsitaB**
Resolves #367

### CODE REVIEW

File: src/Pages/MembershipApplication/register-page.css [hunk 1]
```diff
@@ -1,93 +1,139 @@
...
+ background-color: #8fdcff;
+ border-color: black;
+ color: black;
 }
+
+#login {
+ width: 100%;
+ display: inline-block;
+ text-align: center;
+}
```
Comment (ThaiQ) [inline src/Pages/MembershipApplication/register-page.css:139]: that means add an empty line at the end of the file . [Why?](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file)
image

File: src/Pages/Login/Login.js [hunk 2]
```diff
@@ -37,20 +65,24 @@ export default function Login(props) {
...
-        </button>
-        <p id='SignUp'>
-          <Link to='/register'>Create an account</Link>
-        </p>
-      </div>
-    </form>
+    <Container fluid style={semesterImage}>
+      <Row className='form-card-login'>
+        <form onSubmit={handleSubmit}>
+          <div>
```
Comment (ThaiQ) [inline src/Pages/Login/Login.js:71]: do you use this ```<div>``` ?

File: src/Pages/Login/login.css [hunk 3]
```diff
@@ -69,6 +54,27 @@
...
+  
+  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.26);
+  border-radius: 20px;
+  padding: 1.5rem;
+  background: white;
+  flex: none;
+
+  margin: 0;
+  justify-content: center;
+}
```
Comment (ThaiQ) [inline src/Pages/Login/login.css:80]: That means to add an empty line at the end of the file. [Why?](https://stackoverflow.com/questions/2287967/why-is-it-recommended-to-have-empty-line-in-the-end-of-a-source-file)

image

File: src/Pages/Login/Login.js [hunk 4]
```diff
@@ -23,6 +37,20 @@ export default function Login(props) {
...
+        backgroundImage: `url(${fall})`
+      });
+    }
+  }
+
+  useEffect(() => {
+    getSemesterImage();
+  });
+
   const fields = [
```
Comment (ThaiQ) [inline src/Pages/Login/Login.js:54]: have variable stays together (under line 24)

**Comment by ThaiQ** [comment]
also attach screen-shots for UI/UX PRs; its easier for us

**Comment by IsitaB** [comment]
sce-login
sce-register

File: src/Pages/Login/Login.js [hunk 5]
```diff
@@ -1,13 +1,39 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { Row, Container } from 'reactstrap';
 import { Link } from 'react-router-dom';
 import LoginInput from './LoginInput';
 import { loginUser } from '../../APIFunctions/User';
 import './login.css';
```
Comment (evanugarte) [inline src/Pages/Login/Login.js:7]: remove blank line

File: src/Pages/Login/Login.js [hunk 6]
```diff
@@ -23,34 +49,37 @@ export default function Login(props) {
...
-        })}
-        <button type='submit' id='loginBtn'>
-          Login
-        </button>
-        <p id='SignUp'>
-          <Link to='/register'>Create an account</Link>
-        </p>
-      </div>
-    </form>
+    <Container fluid style={semesterImage}>
```
Comment (evanugarte) [inline src/Pages/Login/Login.js:67]: i like the idea of dynamically rendering the style. However the only thing that changes is `backgroundImage`.

Can you:
1. Move all of the non-changing css to a `.css` file?
1. Do sum like
```jsx
<Container fluid class="your-classname-here" style={{backgroundImage: getSemesterImage()}}>
```
3. Then, change your `getSemesterImage` function to just return the proper image like
```js
function getSemesterImage() {
  if(something) {
    return `url(${fall})`
  }  else {
    return /*the other thing*/
  }
}
```

File: src/Pages/MembershipApplication/MembershipForm.js [hunk 7]
```diff
@@ -1,12 +1,15 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import './register-page.css';
-import { Container, Row, FormGroup, Label, Input, Button } from 'reactstrap';
+import { Row, FormGroup, Input, Button, Container } from 'reactstrap';
 import { memberApplicationState, memberShipPlanToString } from '../../Enums';
 import MajorDropdown from './MajorDropdown';
 import { registerUser, checkIfUserExists } from '../../APIFunctions/User';
 import { sendVerificationEmail } from '../../APIFunctions/Profile';
 import GoogleRecaptcha from './GoogleRecaptcha';
```
Comment (evanugarte) [inline src/Pages/MembershipApplication/MembershipForm.js:9]: remove blank line

File: src/Pages/MembershipApplication/MembershipForm.js [hunk 8]
```diff
@@ -52,6 +55,16 @@ export default function MembershipForm(props) {
       handleChange: e => setPassword(e.target.value)
     }
   ];
+  const [semesterImage, setSemesterImage] = useState({
```
Comment (evanugarte) [inline src/Pages/MembershipApplication/MembershipForm.js:58]: same thing with semester image here as the other comment i left. If you see yourself reusing the same code, consider moving the `getSemesterImage` function to another file in `util` or `APIFunctions`.

File: src/Pages/MembershipApplication/MembershipForm.js [hunk 9]
```diff
@@ -83,69 +96,93 @@ export default function MembershipForm(props) {
...
+          {nameFields.map((input, index) => {
+            return (
+              <FormGroup key={index}>
                 <Input
-                  className='membership-input email-input'
+                  className='name-input membership-input'
                   type={input.type}
                   onChange={input.handleChange}
                   id={input.id}
+                  placeholder={`${input.label}*`}
```
Comment (evanugarte) [inline src/Pages/MembershipApplication/MembershipForm.js:133]: i like this, nice!

File: src/Pages/MembershipApplication/register-page.css [hunk 10]
```diff
@@ -1,93 +1,139 @@
...
+
+.form-card {
+ position: relative;
+ left: 50%;
+ top: 50%;
+ transform: translate(-50%,-50%);
+
+ width: 35%;
+ min-width: 400px;
+
```
Comment (evanugarte) [inline src/Pages/MembershipApplication/register-page.css:13]: remove blank line

File: src/Pages/MembershipApplication/register-page.css [hunk 11]
```diff
@@ -1,93 +1,139 @@
...
+.membership {
+  width: 100%;
+}
+
+.form-card {
+ position: relative;
+ left: 50%;
+ top: 50%;
+ transform: translate(-50%,-50%);
+
```
Comment (evanugarte) [inline src/Pages/MembershipApplication/register-page.css:10]: remove blank line

**Comment by ThaiQ** [comment]
I can still submit with an invalid email. also Travis is failing

**Comment by ThaiQ** [comment]
Rebase to the lastest dev-commit to fix grpc.
@Annabel1054 try to get this done before final. If you can't then after final; we need this for next semester

File: src/Pages/MembershipApplication/MembershipForm.js (hunk 8 above)
Comment (IsitaB) [reply src/Pages/MembershipApplication/MembershipForm.js]: I just changed it to the tech background that Annabel found

**Comment by IsitaB** [comment]
@seemavora I changed the background to Annabel's tech background in this PR

**Comment by IsitaB** [comment]
@ThaiQ I did not mess with that code. I don't think that MembershipForm or the post request handle invalid emails, but I could code that.

**Comment by IsitaB** [comment]
The backgrounds have been changed to match Annabel's, and I changed the buttons' color and hover css to match Nanar's new header. The email validation fix is going to be a new issue.

**Comment by IsitaB** [comment]
Screen Shot 2020-05-22 at 7 21 17 PM
Screen Shot 2020-05-22 at 7 21 26 PM

File: src/Pages/Login/login.css [hunk 12]
```diff
@@ -1,34 +1,22 @@
...
-
-  position: relative;
-  left: 50%;
-  top: 50%;
-  transform: translate(-50%,-5%);
-}
-
-#body h1
-{
+#body h1 {
```
Comment (evanugarte) [inline src/Pages/Login/login.css:1]: you can remove these css rules now

**Pull request merged**